libraries/GCS_MAVLink/MAVLink_routing.cpp

```diff
@@ -94,8 +94,30 @@ detect a reset of the flight controller, which implies a reset of its
 routing table.
 
 */
-bool MAVLink_routing::check_and_forward(GCS_MAVLINK &in_link, const mavlink_message_t &msg)
+bool MAVLink_routing::check_and_forward(uint8_t framing_status,
+                                        GCS_MAVLINK &in_link,
+                                        const mavlink_message_t &msg)
 {
+    // handle the case of loopback of our own messages, due to
+    // incorrect serial configuration.
+    if (msg.sysid == mavlink_system.sysid &&
+        msg.compid == mavlink_system.compid) {
+        return false;  // do not process locally
+    }
+
+    switch (framing_status) {
+    case MAVLINK_FRAMING_OK:
+        break;
+    case MAVLINK_FRAMING_BAD_CRC:
+        if (in_link.option_enabled(GCS_MAVLINK::Option::FORWARD_BAD_CRC)) {
+            forward(in_link, msg);
```

Comment:
it looks like this will bypass the special handling for RADIO_STATUS, so if a RADIO_STATUS message has a bad CRC then it will be forwarded, not likely, but the key if we need to do the special message handling before this check